test/Microsoft.Extensions.Caching.Memory.Tests/TokenExpirationTests.cs

```diff
@@ -187,5 +189,65 @@ public void AddExpiredTokenPreventsCaching()
             result = cache.Get(key);
             Assert.Null(result); // It wasn't cached
         }
+
+        [Fact]
+        public void TokenExpiresOnRegister()
+        {
+            var cache = CreateCache();
+            var key = "myKey";
+            var value = new object();
+            var callbackInvoked = new ManualResetEvent(false);
+            var expirationToken = new TestToken(callbackInvoked);
```

Comment:
Nope, this token calls the callback on register which does the detach token stuff and nulls the ExpirationToken list then returns to the part where it adds to the ExpirationToken list.